assets/js/googlesitekit/datastore/site/info.js

```diff
@@ -508,6 +508,23 @@ export const selectors = {
 	 * @return {(boolean|undefined)} `true` if the Web Stories plugin is enabled, `false` if not. Returns `undefined` if not loaded.
 	 */
 	isWebStoriesActive: getSiteInfoProperty( 'webStoriesActive' ),
+
+	/**
+	 * Determines whether the provided URL matches reference site URL or not.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} url The URL to compare with the reference site URL.
+	 * @return {boolean} TRUE if the URL matches reference site URL, otherwise FALSE.
+	 */
+	isSiteURLMatch: createRegistrySelector( ( select ) => ( state, url ) => {
+		const referenceURL = select( STORE_NAME ).getReferenceSiteURL();
+		const normalizeURL = ( incomingURL ) => incomingURL
+			.replace( /^https?:\/\/(www\.)?/i, '' ) // Remove protocol and optional "www." prefix from the URL.
+			.replace( /\/$/, '' ); // Remove trailing slash.
```

Comment:
The ACs reference `Module::permute_site_url` for the behavior here (although "similar" to it). That method does not consider the path of the site URL although this function will as-is since only a trailing slash is stripped. I'm not sure if that's intended.

@felixarntz should this selector consider the site URL's path at all in the matching or only hostname and scheme variants?

Reply:
@aaemnnosttv Good catch! This is actually wrong in `Module::permute_site_url`, but correct here. We _do_ need to consider a potential path in the site URL.

For the current behavior of the plugin, this only causes a problem for the site verification logic in the non-proxy setup flow, so that is not a high priority. Elsewhere that we use `Module::permute_site_url` we are only using it to get to hostnames anyway, so there it doesn't make a difference either way.

But I'll open an issue to fix this method, since we should have a unified approach to this. Long story short, the implementation _here_ is good.

Reply:
Opened https://github.com/google/site-kit-wp/issues/3233